lib/table/index.js

```diff
@@ -0,0 +1,103 @@
+import { normalizeRow, normalizeTable } from './normalize';
+import { applyHeights, measure } from './size';
+import { renderRow } from './render';
+
+class PDFTable {
+  /**
+   * @param {PDFDocument} document
+   * @param {Table} [opts]
+   */
+  constructor(document, opts = {}) {
+    this.document = document;
+    this.opts = Object.freeze(opts);
+
+    normalizeTable.call(this);
+
+    this._currRowIndex = 0;
+    this._ended = false;
+
+    // Render cells if present
+    if (opts.data) {
+      if (Symbol.asyncIterator in opts.data) {
```

Comment:
Its adding an async api to an otherwise sync api. 

I would avoid adding async until a properly async api is designed for pdfkit so its consistent.

There's also the problem of generating bogus pdf. Earlier in pdfkit that was a lot of async code leading to all kind of corruption. My initial contributions was to remove those async code paths

Alternatively is possible to expose an async function `resolveTableData` that normalize the data including async one

Reply:
we could remove it, although not really sure what harm its doing?

Reply:
> we could remove it, although not really sure what harm its doing?

At API design level is introducing an async behavior to an otherwise synchronous api

In practical, it can occur race conditions. Still occurs but previously was worse leading to all kind of corruption (#1038 #265 #762 #1310 #861)

I did not test if the below would work
```
doc.table(asyncOrSyncData)
doc.text('Text below table?')
```

Reply:
Ah i see your point, coming from typescript where it is explicitly defined as being a promise i guess removes the ambiguity.

I'll remove the async support

Reply:
removed in latest commit